### PR TITLE
NRG: Use configured sync intervals for group stores

### PR DESCRIPTION
**Pull request by neilalexander**
Signed-off-by: Neil Twigg <neil@nats.io>

### CODE REVIEW

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -2078,9 +2087,19 @@ func (js *jetStream) createRaftGroup(accName string, rg *raftGroup, storage Stor
 	storeDir := filepath.Join(js.config.StoreDir, sysAcc.Name, defaultStoreDirName, rg.Name)
 	var store StreamStore
 	if storage == FileStorage {
+		// If the server is set to sync always, do the same for the Raft log.
+		js.srv.optsMu.RLock()
+		syncAlways := js.srv.opts.SyncAlways
+		syncInterval := js.srv.opts.SyncInterval
```
Comment (derekcollison) [inline server/jetstream_cluster.go:2093]: Does this match 5 mins? Noticed this was slightly more delayed then default 2m.

File: server/jetstream_cluster.go [hunk 2]
```diff
@@ -782,9 +782,18 @@ func (js *jetStream) setupMetaGroup() error {
...
+	js.srv.optsMu.RUnlock()
 	fs, err := newFileStoreWithCreated(
-		FileStoreConfig{StoreDir: storeDir, BlockSize: defaultMetaFSBlkSize, AsyncFlush: false, srv: s},
-		StreamConfig{Name: defaultMetaGroupName, Storage: FileStorage},
+		FileStoreConfig{
+			StoreDir: storeDir, BlockSize: defaultMetaFSBlkSize, AsyncFlush: false,
+			SyncAlways: syncAlways, SyncInterval: syncInterval, srv: s,
+		},
+		StreamConfig{
+			Name: defaultMetaGroupName, Storage: FileStorage,
```
Comment (derekcollison) [inline server/jetstream_cluster.go:795]: nit can we go back to 1 line for these?

File: server/jetstream_cluster.go (hunk 1 above)
Comment (neilalexander) [reply server/jetstream_cluster.go:2093]: They will now match the default 2 minutes. There's nothing in the git history that explains why it was 5 minutes.

File: server/jetstream_cluster.go (hunk 2 above)
Comment (neilalexander) [reply server/jetstream_cluster.go]: Done.

File: server/jetstream_cluster.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_cluster.go:2088]: We probably would like to not have them firing at the same time.

Comment (neilalexander) [reply server/jetstream_cluster.go:2088]: I've pushed another commit with an idea, curious on your thoughts: what if the stores used a random interval between 0 and `fs.fcfg.SyncInterval` for the first timer fire, and then tick along on `fs.fcfg.SyncInterval` after that? That way stores syncing should be spread out more.

File: server/filestore.go [hunk 3]
```diff
@@ -7946,7 +7947,11 @@ func (fs *fileStore) setSyncTimer() {
 	if fs.syncTmr != nil {
 		fs.syncTmr.Reset(fs.fcfg.SyncInterval)
 	} else {
-		fs.syncTmr = time.AfterFunc(fs.fcfg.SyncInterval, fs.syncBlocks)
+		// First time this fires will be any time up to the fs.fcfg.SyncInterval,
+		// so that different stores are spread out, rather than having many of
+		// them trying to all sync at once, causing blips and contending dios.
+		start := time.Duration(mrand.Int64N(int64(fs.fcfg.SyncInterval)))
```
Comment (derekcollison) [inline server/filestore.go:7953]: This could return 0, should we add a fixed floor?

Comment (neilalexander) [reply server/filestore.go:7953]: Sure, will do.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**